scripts/wnpp-check.1
.TH WNPP-CHECK 1 "Debian Utilities" "DEBIAN" \" -*- nroff -*-
.SH NAME
wnpp-check \- check if a package is being packaged or if this has been requested
.SH SYNOPSIS
\fBwnpp-check\fR [\fB\-\-exact\fR] \fIpackage\fR ...
.br
\fBwnpp-check \-\-help\fR|\fB\-\-version\fR
.SH DESCRIPTION
\fBwnpp-check\fR downloads the lists of packages which are listed as being
packaged (ITPed) or for which a package has been requested (RFPed) from the
WNPP website and lists any packages supplied on the command line which appear
in those lists.
.PP
Note that WNPP, and therefore \fBwnpp-check\fR's output, is source
package based.
.SH OPTIONS
.TP
.BR \-\-exact ", " \-e
Require an exact package name match, rather than the default substring match.
.TP
.BR \-\-help ", " \-h
Show a summary of options.
.TP
.BR \-\-version ", " \-v
Show version and copyright information.
.SH SEE ALSO
https://www.debian.org/devel/wnpp
.SH EXIT STATUS
.TP
0
None of the packages supplied has an open ITP or RFP
.TP
1
Either an error occurred or at least one package supplied has an open ITP or
RFP
.SH AUTHOR
\fBwnpp-check\fR was written by David Paleino <d.paleino@gmail.com>; this man
page was written by Adam D. Barratt <adam@adam-barratt.org.uk> for the
devscripts package.
\fBwnpp-check\fR was originally based on \fBwnpp-alert\fR, which was written
by Arthur Korn <arthur@korn.ch> and modified by Julian Gilbey <jdg@debian.org>
for the devscripts package.  Both scripts are in the public domain.


debian/NEWS
devscripts (2.16.10) unstable; urgency=medium

  debuild's target mode (i.e. where it runs the specified debian/rules
  targets) now requires a '--' between the debuild options and the targets
  that need to be run.

  For example, "debuild -d -- build-arch binary-arch".

 -- James McCoy <jamessan@debian.org>  Thu, 24 Nov 2016 14:25:12 -0500

devscripts (2.14.0) unstable; urgency=medium

  uscan now looks for upstream's keyring under debian/upstream/ as either
  signing-key.pgp or, for an armored keyring, signing-key.asc.  The previously
  used location, debian/upstream-signing-key.pgp, will be checked as a last
  resort for a transition period.

 -- James McCoy <jamessan@debian.org>  Sat, 25 Jan 2014 10:19:02 -0500

devscripts (2.11.9) unstable; urgency=low

  The default for mk-build-deps --tool option/MKBUILDDEPS_TOOL configuration
  changed from "apt-get" to "apt-get --no-install-recommends".

 -- James McCoy <jamessan@debian.org>  Sun, 17 Jun 2012 22:56:26 +0200

devscripts (2.11.7) unstable; urgency=low

  The heuristic used to determine what action to perform when no options are
  given to debchange now defaults to 'changelog'.  To stay with the previous
  behavior, DEBCHANGE_RELEASE_HEURISTIC=log can be added to the configuration
  file.

  Enabling the changelog heuristic also enables DEBCHANGE_MAINTTRAILER.  This
  can be disabled by adding DEBCHANGE_MAINTTRAILER=no to the configuration
  file.

 -- James McCoy <jamessan@debian.org>  Sun, 29 Apr 2012 18:38:46 -0400

devscripts (2.11.2) unstable; urgency=low

  Support for pgp-style commands has been removed from debsign/debrsign.  As
  such, the DEBSIGN_SIGNLIKE configuration variable has also been removed.

 -- James McCoy <jamessan@debian.org>  Tue, 15 Nov 2011 21:36:40 -0500

devscripts (2.10.54) unstable; urgency=low

    manpage-alert now, as originally intended, prints out the full path to the
    binary which is lacking a manpage instead of simply the binary name.

    Before:
      No manual entry for somebinary

    After:
      No manual entry for /usr/bin/somebinary

 -- James Vega <jamessan@debian.org>  Thu, 06 Aug 2009 18:03:41 -0400

devscripts (2.10.24) unstable; urgency=low

    Starting with devscripts 2.10.21, debchange attempts to
    automatically determine whether an NMU changelog should be
    started, by comparing DEBEMAIL and DEBFULLNAME to the
    package's Maintainer and Uploaders fields.

    This behaviour may now be disabled using the DEBCHANGE_AUTO_NMU
    configuration file variable.

 -- Adam D. Barratt <adam@adam-barratt.org.uk>  Mon, 07 Apr 2008 14:38:08 +0100

devscripts (2.10.20) unstable; urgency=low

    devscripts 2.10.19 modified debcommit's --release option to use the
    uncommitted changes (if any) in the changelog to derive the commit
    message.

    This behaviour is now optional and can be enabled using the -R /
    --release-use-changelog command line option or by setting the
    DEBCOMMIT_RELEASE_USE_CHANGELOG configuration option to "yes".

    The default is to use the previous (i.e. pre-2.10.19) behaviour.

 -- Adam D. Barratt <adam@adam-barratt.org.uk>  Wed, 19 Mar 2008 20:31:20 +0000

devscripts (2.10.19) unstable; urgency=low

    In devscripts 2.10.17, dget was modified to unpack source packages by
    default.  The new -d switch or the DGET_UNPACK configuration file option
    may be used to indicate that the unpacking should not occur.

    As linda has been removed from both unstable and testing, support for
    it has been removed from debuild and cvs-debuild.

 -- Adam D. Barratt <adam@adam-barratt.org.uk>  Thu, 13 Mar 2008 19:33:38 +0000


scripts/wnpp-check.sh
#!/bin/bash

# wnpp-check -- check for software being packaged or requested

# This script is in the PUBLIC DOMAIN.
# Authors:
# David Paleino <d.paleino@gmail.com>
#
# Adapted from wnpp-alert, by Arthur Korn <arthur@korn.ch>

set -e

CURLORWGET=""
GETCOMMAND=""
EXACT=0
PROGNAME=${0##*/}

usage () { echo \
"Usage: $PROGNAME <package name> [...]
  -e,--exact         Require an exact package name match,
                     rather than the default substring match.

  -h,--help          Show this help message
  -v,--version       Show a version message

  Check whether a package is listed as being packaged (ITPed) or has an
  outstanding request for packaging (RFP) on the WNPP website
  https://www.debian.org/devel/wnpp/"
}

version () { echo \
"This is $PROGNAME, from the Debian devscripts package, version ###VERSION###
This script is in the PUBLIC DOMAIN.
Authors: David Paleino <d.paleino@gmail.com>
Adapted from wnpp-alert, by Arthur Korn <arthur@korn.ch>,
with modifications by Julian Gilbey <jdg@debian.org>"
}

TEMP=$(getopt -n "$PROGNAME" -o 'hve' \
	      -l 'help,version,exact' \
	      -- "$@") || (rc=$?; usage >&2; exit $rc)

eval set -- "$TEMP"

while true
do
    case "$1" in
	-h|--help) usage; exit 0 ;;
	-v|--version) version; exit 0 ;;
	-e|--exact) EXACT=1 ;;
	--) shift; break ;;
    esac
    shift
done

if [ -z "$1" ]; then
    usage
    exit 1
fi

PACKAGES="$@"

if command -v wget >/dev/null 2>&1; then
    CURLORWGET="wget"
    GETCOMMAND="wget -q -O"
elif command -v curl >/dev/null 2>&1; then
    CURLORWGET="curl"
    GETCOMMAND="curl -qfsL -o"
else
    echo "$PROGNAME: need either the wget or curl package installed to run this" >&2
    exit 1
fi

WNPP=$(mktemp --tmpdir wnppcheck-wnpp.XXXXXX)
WNPPTMP=$(mktemp --tmpdir wnppcheck-wnpp.XXXXXX)
WNPP_PACKAGES=$(mktemp --tmpdir wnppcheck-wnpp_packages.XXXXXX)
trap 'rm -f "$WNPP" "$WNPPTMP" "$WNPP_PACKAGES"' EXIT

# Here's a really sly sed script.  Rather than first grepping for
# matching lines and then processing them, this attempts to sed
# every line; those which succeed execute the 'p' command, those
# which don't skip over it to the label 'd'

$GETCOMMAND $WNPPTMP https://www.debian.org/devel/wnpp/being_packaged || \
    { echo "$PROGNAME: $CURLORWGET https://www.debian.org/devel/wnpp/being_packaged failed." >&2; exit 1; }
sed -ne 's/.*<li><a href="https\?:\/\/bugs.debian.org\/\([0-9]*\)">\([^:<]*\)[: ]*\([^<]*\)<\/a>.*/ITP \1 \2 -- \3/; T d; p; : d' $WNPPTMP > $WNPP

$GETCOMMAND $WNPPTMP https://www.debian.org/devel/wnpp/requested || \
    { echo "$PROGNAME: $CURLORWGET https://www.debian.org/devel/wnpp/requested failed." >&2; exit 1; }
sed -ne 's/.*<li><a href="https\?:\/\/bugs.debian.org\/\([0-9]*\)">\([^:<]*\)[: ]*\([^<]*\)<\/a>.*/RFP \1 \2 -- \3/; T d; p; : d' $WNPPTMP >> $WNPP

awk -F' ' '{print "("$1" - #"$2") https://bugs.debian.org/"$2" "$3}' $WNPP | sort -k 5 > $WNPP_PACKAGES

FOUND=0
for pkg in $PACKAGES
do
    if [ $EXACT != 1 ]; then
	grep $pkg $WNPP_PACKAGES && FOUND=1
    else
	grep " $pkg$" $WNPP_PACKAGES && FOUND=1
    fi
done

exit $FOUND
